examples/classic_controllers/classic_controllers.py

```diff
@@ -704,41 +902,507 @@ def control(self, state, reference):
                         state[self.i_sd_idx] * self.mp['l_d'] * self.limit[self.u_sd_idx] + self.psi_p) / self.limit[
                          self.u_sq_idx] * self.limit[self.omega_idx]
 
-            u_sd = self.d_controller.control(state[self.i_sd_idx], reference[self.ref_d_idx]) + self.u_sd_0
+            if self.torque_control:
+                u_sd = self.d_controller.control(state[self.i_sd_idx], self.ref[1]) + self.u_sd_0
+            else:
+                u_sd = self.d_controller.control(state[self.i_sd_idx], reference[self.ref_d_idx]) + self.u_sd_0
+
             u_sq = self.q_controller.control(state[self.i_sq_idx], self.ref[0]) + self.u_sq_0
 
             action_temp = self.backward_transformation((u_sq, u_sd), epsilon_d)
             action_temp = action_temp - 0.5 * (max(action_temp) + min(action_temp))
 
             action = np.clip(action_temp, self.action_space.low[0], self.action_space.high[0])
             if action.all() == action_temp.all():
-                self.d_controller.integrate(state[self.i_sd_idx], reference[self.ref_d_idx])
+                if self.torque_control:
+                    self.d_controller.integrate(state[self.i_sd_idx], self.ref[1])
+                else:
+                    self.d_controller.integrate(state[self.i_sd_idx], reference[self.ref_d_idx])
                 self.q_controller.integrate(state[self.i_sq_idx], self.ref[0])
 
         else:
-            ref_abc = self.backward_transformation((self.ref[0], reference[self.ref_d_idx]), epsilon_d)
+            if self.torque_control:
+                ref_abc = self.backward_transformation((self.ref[0], self.ref[1]), epsilon_d)
+            else:
+                ref_abc = self.backward_transformation((self.ref[0], reference[self.ref_d_idx]), epsilon_d)
             action = 0
             for i in range(3):
                 action += (2 ** (2 - i)) * self.abc_controller[i].control(state[self.i_abc_idx[i]], ref_abc[i])
 
+        self.plot(self.external_ref_plots, self.state_names)
         return action
 
-    def plot(self):
-        return dict(ref_state=self.ref_state_idx[:-1], ref_value=self.ref[:-1], color=self.color)
+    def set_ref(self):
+        return dict(ref_state=self.ref_state_idx[:-1], ref_value=self.ref[:-1])
 
     def reset(self):
-        self.overlayed_controller.reset()
+        for overlayed_controller in self.overlayed_controller:
+            overlayed_controller.reset()
         if self.controller_type:
             self.d_controller.reset()
-            for q_controller in self.q_controller:
-                q_controller.reset()
+            self.q_controller.reset()
+
         else:
             for abc_controller in self.abc_controller:
                 abc_controller.reset()
 
+        if self.torque_control:
+            self.torque_controller.reset()
+
+class MTPC:
```

Comment:
I feel that the class name 'MTPC' is not fitting well to the actual functionality, since the class additionally incorporates the MTPF characteristics for flux weakening mode. A better name would ne 'OPS' (for Operationg point selection) or 'TorqueToCurrentConversion' or similar.